client/components/QuestionsAnswers/QuestionsAnswers.jsx

```diff
@@ -1,13 +1,264 @@
 import React from 'react';
-import { COLORS } from '../../settings/colors';
 import styled from 'styled-components';
+import { faSearch } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import axios from 'axios';
+import api from '../../api.js';
 
-const QuestionsAnswers = () => (
-  <Header>Q&A Module</Header>
-);
+
+// const AnswerTitle = styled.div`
+//   display: inline-block
+// `
+
+// const AnswerBody = styled.div`
+//   display: inline-block
+// `
+
+const Wrapper = styled.div`
+  display: flex
+
+`;
+
+const Photos = styled.div`
+  float: left
+`;
+
+const ByUser = styled.div`
+  clear: both
+`;
+
+const QuestionsHidden = styled.div`
+  display: none
+`;
+
+const AnswersHidden = styled.div`
+  display: none
+`;
+
+const AlignRight = styled.span`
+  float: right
+`;
+
+const YesLink = styled.u`
+
+`;
+
+
+
+class QuestionsAnswers extends React.Component {
+  constructor(props) {
+    super(props);
+    this.addHelpful = this.addHelpful.bind(this);
+
+    this.state = {
+      answers: [],
+      questions: []
+    };
+  }
+
+  componentDidMount() {
+    api.getQuestions({id: 37323, page: 1, count: 100})
+      .then((res) => {
+        this.setState({
+          questions: res.results
+        });
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }
+
+
+  // getAnswers(question_id) {
+  //   api.getAnswers({question_id: question_id, page: 1, count: 10})
+  //     .then((res) => {
+  //       console.log(res)
+  //       this.setState({
+  //         answers: res.results
+  //       })
+  //     })
+  //     .catch((err) => {
+  //       console.log(err);
+  //     })
+  // }
+
+
+
+
+  addHelpful() {
+    //to change
+
+  }
+
+  render() {
```

Comment:
There's a lot of logic living in the render() method.  Ideally we should keep the render function as pure as possible and move this logic elsewhere.  For example creating a folder in the `lib` directory called dateFunctions and then exporting months there.

Also, we should probably not be defining variables and doing manipulation inside the render function.

Reply:
@asieke  Thank you very much for your comments, its definitely very helpful to me. I started coding since HR bootcamp, so I don't even know what should I do to let my code better or more ideally most of time, all I can think about  just make it runnable. I will try to fix every point you said. Plz give me as many as advises if you would like to, it will helps me to learn and to improve my coding skills, glad that I can group with you.